### ISSUE TITLE
No ip allocated to interface of converted qcow2 image

### ISSUE THREAD

**Issue by sanketIntel** [linka-cloud/d2vm#4]
WheN I get converted qcow2 image, it is having an interface eth0 but no ip address is not allocated to it when we bring up the vm and interface is also in down state by default. Is it an expected behavior?

**Comment by Adphi** [comment]
Yes, this is the expected behavior. You need to create an image with a network management configuration. In the [complete example](https://github.com/linka-cloud/d2vm/tree/v0.0.6/examples/full) netplan is used because it is the standard for ubuntu (since 18.04).
It is not implemented because there are many ways to manage the network configuration: netplan / ifupdown / systemd-networkd... each distribution having its own method. Maybe it will be implemented in the future.

**Comment by Adphi** [comment]
I just added networking support on the branch [run-push](https://github.com/linka-cloud/d2vm/commit/dfa1e04e9d5e35c1692e0bf9e0aef89c752fe69c).
It adds a `--network` flag to the `build` and `convert` commands.

The supported values are:
- Ubuntu: netplan, ifupdown, none
- Debian: netplan, ifupdown, none
- Alpine: ifupdown, none
- CentOS: nothing as network management is already setup on the base image

**Comment by sanketIntel** [comment]
@Adphi thank you for your quick support. 
Now it is getting ip address.
 The only thing I see missing now is nameserver in resolve.conf and hostname. Is is feasible to add support for it?

**Comment by Adphi** [comment]
@sanketIntel you're welcome.

There should be a `resolv.conf` with `nameserver 8.8.8.8`. 

If you are getting the ip address from *dhcp* you should also configure the *dhcp* server to send the the nameservers.

https://github.com/linka-cloud/d2vm/blob/6b62dbfb9a56f8bbad56dd15830d58e5a5c5dc94/builder.go#L306-L311

I will make the `hostname` and `nameservers` configurable through flags in the next few days:
```
-h, --hostname string                Container host name
    --dns list                       Set custom DNS servers
    --dns-option list                Set DNS options
    --dns-search list                Set custom DNS search domains
    --domainname string              Container NIS domain name
```

If you plan to use it on real cloud environments, you may consider creating a *Dockerfile* using your image as the base and installing `cloud-init` in the image.

**Comment by sanketIntel** [comment]
Actually when I boot my vm using tedezed/ubuntu-container-disk:20.0 image it takes nameserver and hostname properly but when I boot VM using d2vm converted image it is not getting nameservers and hostname properly.

**Comment by sanketIntel** [comment]
Hi @Adphi, can you please share when this run-push branch will be merged to main?

**Comment by Adphi** [comment]
Hi @sanketIntel !
I still need to update the documentation and fix some things first, but it should be merged during September.

**Comment by sanketIntel** [comment]
Hi @Adphi, changes in run-push branch are very important for me (especially interface getting ip from dhcp-server and is in UP state). It will be really helpful if they are merged to main by this month.

**Comment by Adphi** [comment]
@sanketIntel the branch has been merged and [v0.1.0 has been pre-released](https://github.com/linka-cloud/d2vm/releases/v0.1.0-rc2).
